packages/light-client/src/borsh.ts

```diff
@@ -0,0 +1,173 @@
+import { Assignable, Enum } from '@near-js/types';
+import { PublicKey } from '@near-js/crypto';
+import BN from 'bn.js';
+
+export class BorshBlockHeaderInnerLite extends Assignable {
+    height: BN;
+    epoch_id: Uint8Array;
+    next_epoch_id: Uint8Array;
+    prev_state_root: Uint8Array;
+    outcome_root: Uint8Array;
+    timestamp: BN;
+    next_bp_hash: Uint8Array;
+    block_merkle_root: Uint8Array;
+}
+
+export class BorshApprovalInner extends Enum {
+    endorsement?: Uint8Array;
+    skip?: BN;
+}
+
+export class BorshValidatorStakeViewV1 extends Assignable {
+    account_id: string;
+    public_key: PublicKey;
+    stake: BN;
+}
+
+export class BorshValidatorStakeView extends Enum {
+    v1?: BorshValidatorStakeViewV1;
+}
+
+export class BorshValidatorStakeViewWrapper extends Assignable {
+    bps: BorshValidatorStakeView[];
+}
+
+export class BorshEmpty extends Assignable {}
+
+export class BorshPartialExecutionStatus extends Enum {
```

Comment:
This is not a comment for this PR, but a general one for the borsh-js design. I think the design is rather inflexible as it requires that serializables are classes, and that makes it very unnatural that we have to use Enum and Assignable.

I'd imagine something like this is more reasonable:

```ts
export type BorshPartialExecutionStatus = (
  {unknown: {}} | {failure: {}} | {successValue: Uint8Array} | {successReceiptId: Uint8Array});

const SCHEMA_BorshEmpty: Schema<{}> = {
    kind: 'struct',
    fields: [],
};

const SCHEMA_BorshPartialExecutionStatus: Schema<BorshPartialExecutionStatus> = {
    kind: 'enum',
    field: 'enum',
    values: [
        ['unknown', SCHEMA_BorshEmpty],
        ['failure', SCHEMA_BorshEmpty],
        ['successValue', ['u8']],
        ['successReceiptId', [32]],
    ],
};

// to deserialize:
const executionStatus = deserialize(SCHEMA_BorshPartialExecutionStatus, buffer);
if ('successValue' in executionStatus) { useBuffer(executionStatus.successValue); }
// to serialize (no need to wrap):
const buffer = serialize(SCHEMA_BorshPartialExecutionStatus, {failure: {}});
```

But that requires changing borsh-js.

Reply:
Yeah, I agree. Was a large initiative and unrelated to my changes though, so I won't be addressing here.